pkg/controller/deliverable/reconciler.go

```diff
@@ -66,22 +70,29 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return r.completeReconciliation(ctx, deliverable, err)
 	}
 
+	log = log.WithValues("delivery", delivery.Name)
+	ctx = logr.NewContext(ctx, log)
+
 	deliveryGVK, err := utils.GetObjectGVK(delivery, r.Repo.GetScheme())
 	if err != nil {
-		return r.completeReconciliation(ctx, deliverable, controller.NewUnhandledError(fmt.Errorf("get object gvk: %w", err)))
+		log.Error(err, "failed to get object gvk for delivery")
+		return r.completeReconciliation(ctx, deliverable, controller.NewUnhandledError(
+			fmt.Errorf("failed to get object gvk for delivery [%s]: %w", delivery.Name, err)))
 	}
 
 	deliverable.Status.DeliveryRef.Kind = deliveryGVK.Kind
 	deliverable.Status.DeliveryRef.Name = delivery.Name
 
 	if !r.isDeliveryReady(delivery) {
 		r.conditionManager.AddPositive(MissingReadyInDeliveryCondition(getDeliveryReadyCondition(delivery)))
-		return r.completeReconciliation(ctx, deliverable, fmt.Errorf("delivery is not in ready state"))
+		log.Info("delivery is not in ready state")
```

Comment:
We pass an error to `completeReconciliation`. Should this be `log.Error`

Reply:
No, because this is a handled error

Reply:
I'll expound a little to share some thoughts. There are many kinds of 'error' (and I'm not happy about it but it's the truth):
* **an exception**:  "This code should not be reached, but no one likes a panic".   **Use**: Log.Error and retry-with-backoff
* **a recoverable error**.. async/external comms usually. **Use**: Log.Error and retry-with-backoff
* **a message** that we consider recoverable by user action:  "I tried to stamp but couldn't" (often these are deeper user validations). **Use**: Status.Conditions (primary form of user comms), Log.Info
* **a message** that we consider recoverable with a retry: "something is not ready yet". **Use**: Status.Conditions, Log.Info

log.Debug is not necessarily tied to errors, but is there to provide noisy context for debugging.

for `retry-with-backoff` we feel that it would also be great to add a Condition for Backoff to communicate with the user. (should be an issue/discussion though)

Reply:
+100 for making the "we're in a retry backoff" state somehow more explicit (ala when you see a Pod in CrashLoopBackoff, you know the retries will start taking longer)